src/qengine/opencl.cpp

```diff
@@ -1637,8 +1589,15 @@ void QEngineOCL::INTBCD(OCLAPI api_call, bitCapInt toMod, const bitLenInt start,
     if (nibbleCount * 4 != length) {
         throw std::invalid_argument("BCD word bit length must be a multiple of 4.");
     }
-    bitCapInt inOutMask = ((1 << length) - 1) << start;
-    bitCapInt otherMask = ((1 << qubitCount) - 1) ^ inOutMask;
+
+    bitCapInt maxPow = intPow(10U, nibbleCount);
+    toMod %= maxPow;
+    if ((length == 0U) || (toMod == 0U)) {
```

Comment:
Do the length parameter check at the top of the function.

Reply:
Done.